libcaf_core/caf/anon_mail.test.cpp

```diff
@@ -0,0 +1,153 @@
+// This file is part of CAF, the C++ Actor Framework. See the file LICENSE in
+// the main distribution directory for license terms and copyright or visit
+// https://github.com/actor-framework/actor-framework/blob/master/LICENSE.
+
+#include "caf/anon_mail.hpp"
+
+#include "caf/test/fixture/deterministic.hpp"
+#include "caf/test/test.hpp"
+
+#include "caf/behavior.hpp"
+#include "caf/event_based_actor.hpp"
+#include "caf/mailbox_element.hpp"
+#include "caf/message_priority.hpp"
+
+using namespace caf;
+using namespace std::literals;
+
+namespace {
+
+WITH_FIXTURE(test::fixture::deterministic) {
+
+TEST("send asynchronous message") {
+  auto dummy = sys.spawn([](event_based_actor*) -> behavior {
+    return {
+      [=](const std::string&) {},
+    };
+  });
+  SECTION("regular message") {
+    anon_mail("hello world").send(dummy);
+    expect<std::string>()
+      .with("hello world")
+      .priority(message_priority::normal)
+      .from(nullptr)
+      .to(dummy);
+  }
+  SECTION("urgent message") {
+    anon_mail("hello world").urgent().send(dummy);
+    expect<std::string>()
+      .with("hello world")
+      .priority(message_priority::high)
+      .from(nullptr)
+      .to(dummy);
+  }
+  SECTION("invalid receiver") {
+    anon_mail("hello world").urgent().send(actor{});
+    check_eq(mail_count(), 0u);
+  }
+}
+
+TEST("send delayed message") {
```

Comment:
We have good coverage because `delay` is implemented via `schedule`, but I would still like seeing a `.schedule(...)` call in the tests, at least for exemplary purposes.